apps/src/studio/studio.js

```diff
@@ -1170,6 +1170,14 @@ Studio.willSpriteTouchWall = function (sprite, xPos, yPos) {
   return Studio.willCollidableTouchWall(sprite, xCenter, yCenter);
 };
 
+Studio.getWallValue = function (row, col) {
+  if (Studio.walls) {
+    return skin[Studio.walls] ? (skin[Studio.walls][row][col] << constants.WallCoordsShift): 0;
+  } else {
+    return Studio.map[row][col] & constants.WallAnyMask;
```

Comment:
The Studio.map version includes the walls values already shifted. So they are simply masked before returning (so we don't return start, finish, and other map data in this function).

Theoretically, we could mask the first case as well, but the `skin[Studio.walls]` array should only have specific row/col wall data in it, so it should be safe.